File: app/src/main/java/sgtmelon/scriptum/app/view/NoteCallback.java
package sgtmelon.scriptum.app.view;

import androidx.annotation.NonNull;
import sgtmelon.scriptum.app.control.SaveControl;
import sgtmelon.scriptum.app.vm.NoteViewModel;

/**
 * Интерфейс общения {@link NoteFragmentParent} и {@link NoteActivity}
 */
public interface NoteCallback {

    void setupFragment(boolean isSave);

    @NonNull
    SaveControl getSaveControl();

    @NonNull
    NoteViewModel getViewModel();

}

File: app/src/main/java/sgtmelon/scriptum/app/screen/callback/note/NoteCallback.kt
package sgtmelon.scriptum.app.screen.callback.note

import sgtmelon.scriptum.app.screen.view.note.NoteActivity
import sgtmelon.scriptum.app.screen.vm.note.NoteViewModel

/**
 * Интерфейс общения [NoteViewModel] с [NoteActivity]
 *
 * @author SerjantArbuz
 */
interface NoteCallback {

    fun showTextFragment(id: Long, isSave: Boolean)

    fun showRollFragment(id: Long, isSave: Boolean)

    fun onPressBackText(): Boolean

    fun onPressBackRoll(): Boolean

    fun finish()

}